Final EDA Update (#24)
Kydoimos97/CapstoneMSBA2020: EDA/Main_EDA.Rmd

```

```{r, results='asis', warning=FALSE, message=FALSE}

print.xtable(xtable(table2, caption = "Amount of Pumps per SiteID"), comment=FALSE)

rm(table2, list_mpds, list_siteid, i, counter, df2)
```

> Figure 8 shows the aggregate sum of sales per day grouped by site_id, or location. There is a clear difference between various sites in the amount of aggregated sales per day. Every location shows differences in seasonality, e.g., 380 vs. 280, and growth, e.g., 280 vs. 517. Table 7 shows the site_id and the matching amount of pumps. There is no apparent difference in the number of pumps and sales in a given day. Seasonality also does not appear to depend on the number of pumps available at the get station, and neither does growth.

Reviewer: ```suggestion
Figure 8 shows the aggregate sum of sales per day grouped by site_id and indicates differences in sales by location. Each location also conveys seasonality (e.g., 380 vs. 280 and 280 vs. 517).
Table 7 shows the site_id and the matching amount of pumps. There is no apparent difference in the number of pumps and sales in a given day. Seasonality also does not appear to depend on the number of pumps available at the get station, and neither does growth.

```

\newpage

# Results

Through the exploratory analysis we performed on our data, we have discovered the following:

> 1.  We cannot currently predict any values correctly without running a more complex analysis. After trying to use a standard GLM algorithm, we found that it could not accurately predict any Sales value. With that in mind, we will be researching the best method for performing other forms of analysis and which platform is best for this use case (i.e., Python or R). Additionally, we will explore different algorithms that can better handle the data provided.

Reviewer: ```suggestion
1. Currently, we cannot predict any values without running a more complex analysis. We tested a standard GLM algorithm and found it could not accurately predict any Sales value. The team will research the best method for performing other forms of analysis and consider which platform is best for this use case (i.e., Python or R) due to SQL friendly package constraints. Additionally, we will explore different algorithms that can better handle the data provided.

```

> 2.  After mapping the supplementary data set to the original, we found that the temperature variable contained 11,799 NA values. We have a few thoughts on how to go about this but want to do some more research to find best practices. Our insights are to possibly use historical temperature data by site_id to fill in the NA values.

Reviewer: ```suggestion
2.  After mapping the supplementary data set to the original, we found the temperature variable contained 11,799 NA values. As mentioned in the EDA, we have a few thoughts on imputing the missing data but want to research to find best practices. Our thoughts are to possibly use historical temperature data by site_id to fill in the NA values.

```

> 3.  Multicollinearity is present in this data set and will need to be resolved for our final analysis. It exists between the max temp and the min temp and the Gross Book Value, and the Square Footage. For the temperature variables, our thoughts are to exclude Min Temp since that generally occurs during the middle of the night, where Max Temp is a more accurate representation of the regular hours of operation. For GBV and sq_footage, our thoughts are to create two separate variables to help standardize the data.

Reviewer: ```suggestion
3.	We will need to resolve the multicollinearity present in this dataset. Multicollinearity exists between the max temp and the min temp and the Gross Book Value, and the Square Footage. We are considering excluding Min Temp for the temperature variables since it generally occurs during the middle of the night. Max Temp is a more accurate representation of the regular hours of operation. For GBV and sq_footage, our thoughts are to create two separate variables to help standardize the data.

```

> 4.  This data set deals with Singularities, where two variables hold precisely the same or close to the same data. Especially Site_ID, Days_open, and current GBV show singularities as they all contain the same information. We need to thoroughly clean this data set and compare variables to show precisely what information it holds and where we can find information that we can predict so we can focus on the most critical information.

Reviewer: ```suggestion
4.  This dataset contains singularities, where two variables hold precisely the same or close to the same data. Site_ID, Days_open, and current GBV show singularities as they all have the same information. We need to thoroughly clean this data set and compare variables to show precisely what information it holds and where we can find information that we can predict so we can focus on the most critical information.

```

> 5.  Aggregating sales data and displaying it overtime shows seasonality. It can be interesting to dive deeper into an aggregated data sets and learn more about the underlying causes of daily sales before we apply this knowledge to feature engineer the data set to work with candy bar specific sales. Additionally, trying to discover trends by creating visualizations of variables over time will allow us to see where predictive power lies within the data set. Finding trends, however, requires a large number of visualizations, and we did not have the time to go over everything yet by the time of handing in this paper.

Reviewer: ```suggestion
5. Aggregating sales data and displaying it overtime shows seasonality. A deeper dive into an aggregated dataset is warranted to learn more about the underlying causes of daily sales before applying this knowledge to feature engineer the data set to work with candy bar specific sales. Additionally, trying to discover trends by creating visualizations of variables over time will allow us to see where predictive power lies within the dataset. 

```

Currently, we cannot derive much insight into the overall predictability of our variables using a general linear model. However, we could find issues within the data that will help us be more prepared for when we perform more complex analyses. This EDA has helped us understand the different approaches we need to take both in pre-processing and our time-series method. As it stands, dealing with multicollinearity and the NA values and focussing on trends that show over time will help us predict our model.

# Team Member Contribution

> Sam, Willem, and Kelsey produced an initial EDA working together. Tim also created an initial EDA, which was compared to the group EDA. Tim initially worked due to seperately due to scheduling issues. We merged the initial elements of both documents. Willem then started a master document by writing all the code for the final EDA and writing an initial write-up. Kelsey, Willem, Sam, and Tim then added more in-depth written analysis. Additionally, Kelsey, Sam, and Willem proofread to the document and added clarifying text. Finals editing was done by Willem. For collaboration, the whole team used [GitHub](https://github.com/Kydoimos97/CapstoneMSBA2020) to make final edits and adjustments to the EDA.

Reviewer: ```suggestion
Sam, Willem, and Kelsey produced an initial EDA working together. Tim also created an initial EDA, which was compared with the group output. Tim initially worked separately due to scheduling issues. We merged the initial elements of both documents. Willem created the master document by writing all the code for the final EDA and writing an initial write-up. Kelsey, Willem, Sam, and Tim then added more in-depth written analysis. Additionally, Kelsey, Sam, and Willem proofread the document and added clarifying text. Finals editing was done by Willem and Kelsey. For collaboration, the whole team used [GitHub](https://github.com/Kydoimos97/CapstoneMSBA2020) to make final edits and adjustments to the EDA.
```
